From 431a426623b2e46a0968d8fc631f36ec0cf7298f Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Sun, 19 Jan 2014 12:59:45 -0800
Subject: [PATCH] Replace 'pointer' with the equivalent 'void *'.

The pointer typedef is being removed because it causes so many
compiler warnings when -Wshadow is enabled.

Signed-off-by: Keith Packard <keithp@keithp.com>
Reviewed-by: Gaetan Nadon <memsize@videotron.ca>
---
 fontproto.h  |  2 +-
 fontstruct.h | 36 ++++++++++++++++++------------------
 2 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/fontproto.h b/fontproto.h
index 1ab8f37..490629e 100644
--- a/fontproto.h
+++ b/fontproto.h
@@ -74,7 +74,7 @@ extern FontPtr  CreateFontRec (void);
 extern void  DestroyFontRec (FontPtr font);
 extern Bool     _FontSetNewPrivate (FontPtr        /* pFont */,
 				    int            /* n */,
-				    pointer        /* ptr */);
+				    void *         /* ptr */);
 extern int      AllocateFontPrivateIndex (void);
 extern void ResetFontPrivateIndex (void);
 
diff --git a/fontstruct.h b/fontstruct.h
index 97e771a..44bed90 100644
--- a/fontstruct.h
+++ b/fontstruct.h
@@ -122,14 +122,14 @@ typedef struct _Font {
     void        (*unload_font) (FontPtr         /* font */);
     void        (*unload_glyphs) (FontPtr         /* font */);
     FontPathElementPtr fpe;
-    pointer     svrPrivate;
-    pointer     fontPrivate;
-    pointer     fpePrivate;
+    void        *svrPrivate;
+    void        *fontPrivate;
+    void        *fpePrivate;
     int		maxPrivate;
-    pointer	*devPrivates;
+    void        **devPrivates;
 }           FontRec;
 
-#define FontGetPrivate(pFont,n) ((n) > (pFont)->maxPrivate ? (pointer) 0 : \
+#define FontGetPrivate(pFont,n) ((n) > (pFont)->maxPrivate ? (void *) 0 : \
 			     (pFont)->devPrivates[n])
 
 #define FontSetPrivate(pFont,n,ptr) ((n) > (pFont)->maxPrivate ? \
@@ -149,14 +149,14 @@ typedef struct _FontPathElement {
     char       *name;
     int         type;
     int         refcount;
-    pointer     private;
+    void        *private;
 }           FontPathElementRec;
 
 typedef Bool (*NameCheckFunc) (char *name);
 typedef int (*InitFpeFunc) (FontPathElementPtr fpe);
 typedef int (*FreeFpeFunc) (FontPathElementPtr fpe);
 typedef int (*ResetFpeFunc) (FontPathElementPtr fpe);
-typedef int (*OpenFontFunc) ( pointer client,
+typedef int (*OpenFontFunc) ( void *client,
 			      FontPathElementPtr fpe,
 			      Mask flags,
 			      char* name,
@@ -168,55 +168,55 @@ typedef int (*OpenFontFunc) ( pointer client,
 			      char** aliasName,
 			      FontPtr non_cachable_font);
 typedef void (*CloseFontFunc) (FontPathElementPtr fpe, FontPtr pFont);
-typedef int (*ListFontsFunc) (pointer client,
+typedef int (*ListFontsFunc) (void *client,
 			      FontPathElementPtr fpe,
 			      char* pat,
 			      int len,
 			      int max,
 			      FontNamesPtr names);
 
-typedef int (*StartLfwiFunc) (pointer client,
+typedef int (*StartLfwiFunc) (void *client,
 			      FontPathElementPtr fpe,
 			      char* pat,
 			      int len,
 			      int max,
-			      pointer* privatep);
+			      void ** privatep);
 
-typedef int (*NextLfwiFunc) (pointer client,
+typedef int (*NextLfwiFunc) (void *client,
 			     FontPathElementPtr fpe,
 			     char** name,
 			     int* namelen,
 			     FontInfoPtr* info,
 			     int* numFonts,
-			     pointer private);
+			     void *private);
 
 typedef int (*WakeupFpeFunc) (FontPathElementPtr fpe,
 			      unsigned long* LastSelectMask);
 
-typedef void (*ClientDiedFunc) (pointer client,
+typedef void (*ClientDiedFunc) (void *client,
 			       FontPathElementPtr fpe);
 
-typedef int (*LoadGlyphsFunc) (pointer client,
+typedef int (*LoadGlyphsFunc) (void *client,
 			       FontPtr pfont,
 			       Bool range_flag,
 			       unsigned int nchars,
 			       int item_size,
 			       unsigned char* data);
 
-typedef int (*StartLaFunc) (pointer client,
+typedef int (*StartLaFunc) (void *client,
 			    FontPathElementPtr fpe,
 			    char* pat,
 			    int len,
 			    int max,
-			    pointer* privatep);
+			    void ** privatep);
 
-typedef int (*NextLaFunc) (pointer client,
+typedef int (*NextLaFunc) (void *client,
 			   FontPathElementPtr fpe,
 			   char** namep,
 			   int* namelenp,
 			   char** resolvedp,
 			   int* resolvedlenp,
-			   pointer private);
+			   void *private);
 
 typedef void (*SetPathFunc)(void);
 
-- 
1.8.4.2

